src/Connector.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.18;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {ICurvePool} from "./interfaces/ICurvePool.sol";
+import {IStrategyInterface} from "./interfaces/IStrategyInterface.sol";
+
+contract Connector {
+    using SafeERC20 for IERC20;
+
+    address public immutable xvault;
+
+    IStrategyInterface public immutable strategy;
+
+    ICurvePool public immutable curvePool;
+
+    IERC20 public immutable assetA;
+    IERC20 public immutable assetB;
+
+    constructor(address _xvault, address _strategy, address _assetA, address _assetB) { // @todo - add sanity checks
+        xvault = _xvault;
+        strategy = IStrategyInterface(_strategy);
+        assetA = IERC20(_assetA);
+        assetB = IERC20(_assetB);
+        curvePool = ICurvePool(strategy.asset());
+        // @todo - approvals
+    }
+
+    function rate() external view returns (uint256) {
+        // return strategy.rate();
+        // @todo - use pessimistic oracle
```

Comment:
looks like this is the biggest question of the implementation

Did we recognize that we have an issue with the pessimistic oracle that the price fluctuates in terms of our denomination asset?

or was there new info uncovered here?